.github/workflows/lint.yml

```diff
@@ -21,6 +21,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
+      with:
+        fetch-depth: '0'
```

Comment:
@StevenACoffman I try to fix the .golangci.yaml, but it shows many lint issues that this PR hasn't introduced. So I enable the `new` configuration in the issues section, and this would depend on the history commits.